scripts/Crawlers/BaseCrawler.py

```diff
@@ -552,3 +558,22 @@ def _check_dats_present(self, directory):
                         return os.path.join(file_path, subfile_name)
             elif file_name.lower() == "dats.json":
                 return file_path
+
+    def _add_source_data_submodule_if_derivedfrom_conp_dataset(
+        self, dats_json, dataset_dir
+    ):
+        f = open(dats_json)
+        metadata = json.loads(f.read())
+        f.close()
+
+        source_dataset_link = None
+        source_dataset_id = None
+        for property in metadata["extraProperties"]:
+            if property["category"] == "derivedFrom":
+                source_dataset_link = property["values"][0]["value"]
```

Comment:
Is there some guarantee that all of these values are going to be present? The [DATS dataset schema](https://github.com/CONP-PCNO/schema/blob/master/dataset_schema.json#L202) does not require any fields for [`category_values_pair_schema`](https://github.com/CONP-PCNO/schema/blob/master/category_values_pair_schema.json) or the [`annotation_schema`](https://github.com/CONP-PCNO/schema/blob/master/annotation_schema.json) that will be nested in it. So unless these are forced by the crawler somewhere else, this line could raise `KeyError` or `IndexError` unexpectedly here.

Reply:
I just added a few if statements above to check whether each values are present, if not, then go on to the next property. Not sure if there is a more elegant way to check those 3 things...

Reply:
I think this is mostly a style question but you could make this a bit cleaner by using `try` and `except`:

```
try:
    source_dataset_link = property["values"][0]["value"]
except (KeyError, IndexError):
    continue
```

Reply:
Niice!! Done in the next commit. Thank you!